src/common/menu.rs

```diff
@@ -70,6 +87,15 @@ impl<'a> SubMenu<'a> {
             value
         });
     }
+
+    pub fn add_onoffselector(&mut self, title: &'a str, checked: bool) {
+        // TODO: Is there a more elegant way to do this?
+        // The HTML only supports a single onoffselector but the SubMenu stores it as a Vec
```

Comment:
Does it need to be a Vec? It seems you are adding it as a Vec in this PR, what happens if onoffselector is simply of type `OnOffSelector<'a>`?

Reply:
There needs to be three states - On, Off, and None. That way the templating knows whether to add a toggle menu or an onoffselector. Theres a little bit of logic in the template now that looks if the vec is truthy / falsy to decide which type of submenu to add. So maybe an Option<OnOff> would work instead of a vec? I wasn't able to figure that out with the template but I think its a good idea to move away from a vec if we can.